arroyo-controller/src/lib.rs

```diff
@@ -427,6 +434,64 @@ impl ControllerGrpc for ControllerServer {
             Err(err) => Err(Status::from_error(Box::new(err))),
         }
     }
+
+    async fn check_udfs(
+        &self,
+        request: Request<CheckUdfsReq>,
+    ) -> Result<Response<CheckUdfsResp>, Status> {
+        let endpoint = match env::var(REMOTE_COMPILER_ENDPOINT_ENV) {
+            Ok(endpoint) => endpoint,
+            Err(_) => {
+                warn!("Remote compiler is required for checking UDFs, skipping.");
+                return Ok(Response::new(CheckUdfsResp {
+                    result: ValidationResult::Ok as i32,
+                    errors: vec![],
+                }));
+            }
+        };
+
+        let mut client = CompilerGrpcClient::connect(endpoint)
+            .await
+            .map_err(|e| io::Error::new(ErrorKind::Other, format!("{}", e)))?;
+
+        let module = get_udfs_mod(&request.into_inner().udfs)?;
+
+        client
+            .check_udfs_mod(CheckUdfsModReq { udfs_mod: module })
+            .await
+    }
+}
+
+fn get_udfs_mod(udfs: &Vec<String>) -> Result<String, Status> {
+    let imports = quote! {
+        #![allow(warnings)]
+        use types::*;
+        use chrono;
+        use std::time::SystemTime;
+        use std::str::FromStr;
+        use serde::{Deserialize, Serialize};
+    };
```

Comment:
@mwylde I looked at the imports in `compile_pipeline_main` and kind of guessed what we want for udfs. What do you think?

Reply:
I've changed it so the exact contents of the UDFs editor are written to the build directory, so it's now up to the user to write whatever `use` statements they need.